dev/plugins/hu.elte.txtuml.seqdiag/src/hu/elte/txtuml/seqdiag/export/plantuml/exporters/fragments/ParFragmentExporter.java

```diff
@@ -31,18 +45,73 @@ public boolean validElement(ASTNode curElement) {
 	@SuppressWarnings("unchecked")
 	@Override
 	public boolean preNext(MethodInvocation curElement) {
-		List<LambdaExpression> params = (List<LambdaExpression>) curElement.arguments();
+		List<Expression> params = (List<Expression>) curElement.arguments();
 		compiler.println("par");
 		boolean isFirst = true;
-		for (LambdaExpression interaction : params) {
+		for (Expression interaction : params) {
 			if (isFirst) {
 				isFirst = false;
 			} else {
 				compiler.println("else");
 			}
-			interaction.getBody().accept(compiler);
+			switch (interaction.getNodeType()){
+			case ASTNode.LAMBDA_EXPRESSION:
+				((LambdaExpression)(interaction)).getBody().accept(compiler);
+				break;
+			case ASTNode.CLASS_INSTANCE_CREATION:
+				ExportRun(((ClassInstanceCreation)(interaction)).getAnonymousClassDeclaration().bodyDeclarations());				
+				break;
+			case ASTNode.SIMPLE_NAME:
+				SimpleName simpleName = ((SimpleName)(interaction));
+				AbstractTypeDeclaration declarationSimpleName = getTypeDeclaration(simpleName.resolveTypeBinding());
+				if (declarationSimpleName!=null) {
+					ExportRun(declarationSimpleName.bodyDeclarations());
+				}
+				break;
+			case ASTNode.QUALIFIED_NAME:
+				QualifiedName qualifiedName = ((QualifiedName)(interaction));
+				AbstractTypeDeclaration declarationQualifiedName = getTypeDeclaration(qualifiedName.resolveTypeBinding());
+				if (declarationQualifiedName!=null){
+					ExportRun(declarationQualifiedName.bodyDeclarations());
+				}
+				break;
+			}
 		}
 		return false;
 	}
-
+	
+	private static AbstractTypeDeclaration getTypeDeclaration(ITypeBinding binding) {
+		try {
+			CompilationUnit cu = getCompilationUnit(binding);
+			AbstractTypeDeclaration declaration = (AbstractTypeDeclaration) cu.findDeclaringNode(binding.getKey());
+			return declaration;
+		} catch (NullPointerException ex) {
+			return null;
+		}
+	}
+	
+	private static CompilationUnit getCompilationUnit(IBinding binding) {
```

Comment:
If you are using the method this way, there is no reason for it to be a static method.